### PR TITLE
new(libsinsp/test): add basic tests for networking and container events

### PR DESCRIPTION
**Pull request by LucaGuerra**
**What type of PR is this?**

> Uncomment one (or more) `/kind <>` lines:

/kind feature

**Any specific area of the project related to this PR?**

> Uncomment one (or more) `/area <>` lines:

> /area libsinsp

/area tests

> /area proposals

**Does this PR require a change in the driver versions?**

> /version driver-API-version-major

> /version driver-API-version-minor

> /version driver-API-version-patch

> /version driver-SCHEMA-version-major

> /version driver-SCHEMA-version-minor

> /version driver-SCHEMA-version-patch

**What this PR does / why we need it**:

These new tests exercise simple cases of ipv4/ipv6 connections (which we all know can have many weird corner cases), including some highlighted here: https://github.com/falcosecurity/libs/issues/260 , plus container metadata parsing through container events.

More importantly, I think, convenience functions to deal with sockaddr have been added, so that people that need to add more tests don't need to figure out the scap format for connections.

This brings libsinsp coverage from ~19% to  ~23%~ ~24%

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

**Comment by FedeDP** [comment]
/milestone 0.10.0

But this can be moved to 0.9.0 as it adds tests, IMHO.

File: userspace/libsinsp/test/sinsp.ut.cpp [hunk 1]
```diff
@@ -511,6 +513,23 @@ TEST_F(sinsp_with_test_input, execveat_invalid_path)
 	}
 }
 
+// Check that the path in case of execve is correctly overwritten in case it was not possible to collect it from the
+// entry event but it is possible to collect it from the exit event
+TEST_F(sinsp_with_test_input, execve_invalid_path_entry)
```
Comment (FedeDP) [inline userspace/libsinsp/test/sinsp.ut.cpp:518]: I think this can be the right time to split this test file into multiple pieces; wdyt?

**Comment by LucaGuerra** [comment]
Will be reopening this now, with 50% more refactoring and rebasing
